cloud-provider.sh

```diff
@@ -26,10 +26,11 @@ set_azure_config() {
   local az_subnet_name=$(az vm nic show -g ${az_resources_group} --vm-name ${az_vm_name} --nic ${az_vm_nic}| jq -r .ipConfigurations[0].subnet.id| cut -d"/" -f 11)
   local az_vnet_name=$(az vm nic show -g ${az_resources_group} --vm-name ${az_vm_name} --nic ${az_vm_nic}| jq -r .ipConfigurations[0].subnet.id| cut -d"/" -f 9)
   local az_vnet_resource_group=$(az vm nic show -g ${az_resources_group} --vm-name ${az_vm_name} --nic ${az_vm_nic}| jq -r .ipConfigurations[0].subnet.id| cut -d"/" -f 5)
+  local az_vm_nsg=$(az vm nic show -g ${az_resources_group} --vm-name ${az_vm_name} --nic ${az_vm_nic} | jq -r .networkSecurityGroup.id | cut -d "/" -f 9)
 
   az logout 2>&1 > /dev/null
 
-if [ -z "$az_subscription_id" ] || [ -z "$az_location" ] || [ -z "$az_resources_group" ] || [ -z "$az_vnet_resource_group" ] || [ -z "$az_subnet_name" ] || [ -z "$az_vnet_name" ]; then
+if [ -z "$az_subscription_id" ] || [ -z "$az_location" ] || [ -z "$az_resources_group" ] || [ -z "$az_vnet_resource_group" ] || [ -z "$az_subnet_name" ] || [ -z "$az_vnet_name" ] || [ -z "$az_vm_nsg" ]; then
```

Comment:
Sometimes the `az_vm_nsg` would return a valid `null` value. So I would object to this change being required, and added, but only if it exists.

Reply:
Snippet from a VM we have in our infrastructure: 
```
az vm nic show -g ${az_resources_group} --vm-name ${az_vm_name} --nic ${az_vm_nic}
```

```
 ... 

      "subnet": {
        "additionalProperties": {},
        "addressPrefix": null,
        "etag": null,
        "id": "/subscriptions/<redacted>...",
        "ipConfigurations": null,
        "name": null,
        "networkSecurityGroup": null,
        "provisioningState": null,
        "resourceGroup": "<redacted>",
        "resourceNavigationLinks": null,
        "routeTable": null,
        "serviceEndpoints": null
      }
...
```

Reply:
It does evaluate correctly, you're right. Although I wonder if setting the variable to null in the cloud-provider.json would be a problem.

Reply:
@HighwayofLife actually i don't check the name, however i check the id, at the 9th field you will find the security group name, AFAIK every vm has to have a security group

Reply:
```
local az_vm_nsg=$(az vm nic show -g ${az_resources_group} --vm-name ${az_vm_name} --nic ${az_vm_nic} | jq -r .networkSecurityGroup.id | cut -d "/" -f 9)
```

Reply:
Not every VM has to have a security group, almost all of our VMs don't, which is why it shows null in the snippet I posted above. But as long as a literal value of "null" is acceptable in the cloud config, this might be okay. -- For reference info, we're running out of Azure Private Cloud, not Public Cloud, through Express Route to on-prem networks.